### ISSUE TITLE
Lock down lemmy private instance

### ISSUE THREAD

**Issue by EnigmaCurry** [EnigmaCurry/d.rymcg.tech#93]
Lemmy can be made to require registration, but there is a [bug that makes RSS feeds always public](https://github.com/LemmyNet/lemmy/issues/3785)

Introduce a new variable `LEMMY_PRIVATE=true` (default false) that will add Traefik middleware to block the paths starting with `/feeds`, this way it will prevent the private data from the communities from leaking through the RSS feeds.

**Comment by mcmikemn** [comment]
Started `issue#93` branch. The new LEMMY_PRIVATE env var in .env* isn't being seen by ytt - still trying to figure that out, _then_ I'll see if my guess at traefik labels is anywhere near correct. :) Ran out of time at the moment.

**Comment by EnigmaCurry** [comment]
Thanks Mike, the middleware you want to use is [blockpath](https://github.com/traefik/plugin-blockpath) which I have already installed as a plugin. There are currently no projects that are using this plugin, but there was an older version of whoami that I used to use it with. You can see the example on the blockpath README, but notice that the config is written in TOML not using docker labels, you'll have to convert it.

The way the docker-compose.instance.yaml works is that there is a variable called `enabled_middlewares` this is a list that gets added to through the course of the template, and then at the very end there is only ONE line that adds all the middleware to the router. You can't add this line more than once, so its all at the bottom using the `enabled_middlewares` line. So you only need to do the template so that it adds the blockpath to that list. You can see an example of this here where it conditionally adds the basicauth middleware to the `enabled_middlewares` list: https://github.com/EnigmaCurry/d.rymcg.tech/blob/12392b468f918a34e1e8e08ecea53d779249aba6/s3-proxy/docker-compose.instance.yaml#L35-L38

Here is the older example of whoami that uses blockpath, but this is from before I switched to using docker-compose.instance.yaml - https://github.com/EnigmaCurry/d.rymcg.tech/blob/af74417925f8ff96d1a7bce6854cc3cfb19597e5/whoami/docker-compose.yaml#L30-L34

**Comment by mcmikemn** [comment]
I think I got it ([48bb272](https://github.com/EnigmaCurry/d.rymcg.tech/commit/48bb2725751263f14b5a3f542ae6febbd82fcc9e)). When `LEMMY_PRIVATE=true`, `https://<domain>/feeds/c/testcommunity.xml` results in `XML Parsing Error: no root element found`

We may want to make it a cleaner fail (like a 404 or 403), but I'd need guidance on how to do that.

**Comment by EnigmaCurry** [comment]
it looks right, but it doesn't sound like its working. It should already return a 403 Forbidden message, nothing to do with XML (seems like its returning the feed still)

**Comment by mcmikemn** [comment]
Well, it doesn't return the feed. Perhaps the browser is seeing the URL ends in `.xml` and is getting a null result so the _browser_ is saying "no root element"?

**Comment by mcmikemn** [comment]
No, I see that blockpath should by default return a 403. Maybe I configured it wrong, or incompletely. Looking into it.

**Comment by EnigmaCurry** [comment]
try it with curl.

I think you need to edit the template https://github.com/EnigmaCurry/d.rymcg.tech/blob/48bb2725751263f14b5a3f542ae6febbd82fcc9e/lemmy/docker-compose.instance.yaml#L19

this should be a boolean expression try:
```
#@ lemmy_private = data.values.lemmy_private == "true"
```

**Comment by EnigmaCurry** [comment]
when your run make install, it creates the docker_compose.override-INSTANCE.yaml which is the final output of ytt. You can check it see that its doing it right or not.

**Comment by mcmikemn** [comment]
When I have 
```
#@ lemmy_private = data.values.lemmy_private
...
#@ if lemmy_private == "true": 
```
in `docker-compose.instance.yaml` and 
```
LEMMY_PRIVATE=true
```
in `.env_<instance>`, then `make config` does add
```
- traefik.http.middlewares.lemmy-${LEMMY_INSTANCE}-proxy-blockpath.plugin.blockpath.regex=^/feeds.*
- traefik.http.routers.lemmy-${LEMMY_INSTANCE}-proxy.middlewares=lemmy-${LEMMY_INSTANCE}-proxy-ipwhitelist,lemmy-${LEMMY_INSTANCE}-proxy-blockpath
```
to `docker-compose.override_<instance>.yaml`.

If I change in `docker-compose.instance.yaml` to
```
#@ lemmy_private = data.values.lemmy_private == "true"
```
`docker-compose.override_<instance>.yaml` only has
```
- traefik.http.routers.lemmy-${LEMMY_INSTANCE}-proxy.middlewares=lemmy-${LEMMY_INSTANCE}-proxy-ipwhitelist
```

That seems like it's working, right?

**Comment by EnigmaCurry** [comment]
Oh you only need to check the == true one place, like:

#@ lemmy_private = data.values.lemmy_private == "true"
...
#@ if lemmy_private:

**Comment by EnigmaCurry** [comment]
looks right, open a pull request please

**Comment by mcmikemn** [comment]
Fixed in #95
